content/references/rippled-api/api-conventions/serialization.md

```diff
@@ -0,0 +1,285 @@
+
+For example, the `Account` field of a [Payment transaction][] [has sort code 1](https://github.com/ripple/rippled/blob/72e6005f562a8f0818bc94803d222ac9345e1e40/src/ripple/protocol/impl/SField.cpp#L219), so it comes before the `Destination` field which [has sort code 3](https://github.com/ripple/rippled/blob/72e6005f562a8f0818bc94803d222ac9345e1e40/src/ripple/protocol/impl/SField.cpp#L221).
+
+The field code is combined with the type code to make a field's [Field ID](#field-ids).
+
+### Variable-Length Encoding
+
+Some types of fields are Variable-Length encoding, which means they are not always the same byte length and are prefixed with a length indicator to indicate how much data they contain. `Blob` fields (containing arbitrary binary data) are one such variable-length encoded type. For a list of which types are variable-length encoded, see the [Type List](#type-list) table.
+
+**Note:** Some types that are not variable-length encoded nonetheless vary in length. These types have different ways of indicating how long they are.
+
+Variable-length fields are encoded with one to three bytes indicating the length of the field immediately after the type prefix and before the contents.
+
+- If the field contains 0 to 192 bytes of data, the first byte defines the length of the VariableLength data, then that many bytes of data follow immediately after the length byte.
+
+- If the field contains 193 to 12480 bytes of data, the first two bytes indicate the length of the field with the following formula:
+
+        193 + ((byte1 - 193) * 256) + byte2
+
+- If the field contains 12481 to 918744 bytes of data, the first three bytes indicate the length of the field with the following formula:
+
+        12481 + ((byte1 - 241) * 65536) + (byte2 * 256) + byte3
+
+- A variable-length field cannot contain more than 918744 bytes of data.
+
+When decoding, you can tell from the value of the first length byte whether there are 0, 1, or 2 additional length bytes:
+
+- If the first length byte has a value of 192 or less, then that's the only length byte and it contains the exact length of the field contents in bytes.
+- If the first length byte has a value of 193 to 240, then there are two length bytes.
+- If the first length byte has a value of 241 to 254, then there are three length bytes.
+
+
+## Type List
+
+Transaction instructions may contain fields of any of the following types:
+
+| Type Name     | Type Code | Variable-Length? | Description                   |
+|:--------------|:----------|:-----------------|:------------------------------|
+| [AccountID][] | 8         | Yes              | The unique identifier for an [account](accounts.html). This field is variable-length encoded, but always exactly 20 bytes. |
```

Comment:
Just to make sure I'm following - when we say that this is always exactly 20 bytes - we're just clarifying that even if it is variable-length encoded, its length never varies and will always be exactly 20 bytes. Or is 20 bytes not referring to the thing we are saying has a variable length?

Reply:
No, you're correct. It's "variable-length" encoded but the length never varies. I'm pretty sure it's a mistake in the early code that we're stuck with now.